crates/typos-cli/src/file_type.rs

```diff
@@ -107,7 +107,7 @@ impl Types {
     }
 
     pub fn file_matched(&self, path: &std::path::Path) -> Option<&str> {
-        let file_name = path.file_name()?;
+        let file_name = path.file_name()?.to_str()?.trim_end_matches(".in");
         let mut matches = self.matches.get_or_default().borrow_mut();
         self.set.matches_into(file_name, &mut *matches);
```

Comment:
Should we loop over this, checking for matches, in case a user wanted to define a file type for `.in` files?

Reply:
Good point. I think 2f87caf436c56be10407bdced138b698874b8c79 addresses this.